subprojects/core-api/src/main/java/org/gradle/api/configuration/BuildFeature.java

```diff
@@ -19,42 +19,59 @@
  * <p>
- * It is possible to check if the feature is {@link #getActive() active} in the current build.
- * The {@link #getRequested() requested} property shows whether the user opted in or opted out from the feature.
+ * Gradle features can be enabled or disabled via <a href="https://docs.gradle.org/current/userguide/build_environment.html">build options</a>,
+ * but in some cases other flags or features can affect that.
+ * For this reason, the {@link #getRequested()} status represents the configuration of a feature in isolation,
+ * and the {@link #getActive()} status represents the final result, after everything has been taken into account.
+ * <p>
+ * Some flags disable features.
+ * For instance, {@code --export-keys} flag always disables {@link BuildFeatures#getConfigurationCache() Configuration Cache}.
+ * If {@code --configuration-cache} is provided at the same time, then {@code requested == true}, but {@code active == false}.
+ * <p>
+ * Some features automatically enable other features.
+ * For instance, enabling {@link BuildFeatures#getIsolatedProjects() Isolated Projects} enables Configuration Cache.
+ * So that {@code configurationCache.requested == null}, but {@code configurationCache.active == true}.
  *
  * @see BuildFeatures
  * @since 8.5
  */
 public interface BuildFeature {
 
     /**
-     * Whether the feature was requested for the build.
-     * <p>
-     * The provider <b>can be undefined</b> if the user did not explicitly opt in or opt out from a feature.
-     * Use {@link Provider#getOrNull()} to safely retrieve a nullable value or check {@link Provider#isPresent()}.
+     * How the feature was requested:
+     * <ul>
+     *     <li>true - explicitly enabled (e.g. {@code --configuration-cache})
+     *     <li>false - explicitly disabled (e.g. {@code --no-configuration-cache})
+     *     <li>null (value not present) - no preference, default behavior
+     * </ul>
+     *
+     * Use {@link Provider#getOrNull()} to safely retrieve a nullable value or check {@link Provider#isPresent()}
+     * as the provider <b>can be undefined</b> in the case there is no explicit request.
      * <p>
-     * This method is primarily useful for gathering feature usage statistics, as it corresponds to the user intention.
+     * Note that enabling the feature doesn't necessary mean the feature will be activated.
+     * See {@link BuildFeature} for more details.
      * <p>
-     * Note that the requested state does not always imply that the feature is active in the build.
-     * In case an effective status is needed, use {@link #getActive()}.
+     * Use {@link #getActive()} to get the effective status of the feature.
      *
      * @since 8.5
      */
     Provider<Boolean> getRequested();
 
     /**
-     * Whether the feature is active in the build.
-     * <p>
-     * The provider is always defined and its value denotes the effective status of a feature in a build.
```

Comment:
💭 

This bit was important IMO, when dealing with providers, you need to know whether it can have no value. 

Maybe use the `@return` doclet tag to express that

Reply:
Added the sentence as part of `@return`.